### PR TITLE
Align KTO with DPO: Fix `logits/chosen` and `logits/rejected` metrics

### PR DESCRIPTION
**Pull request by qgallouedec**
`KTOTrainer._compute_loss` selected the chosen/rejected logits as

```python
chosen_logits = outputs.logits.index_select(0, chosen_idx)
```

i.e. the **full** `[n_chosen, seq_len, vocab_size]` tensor, and then logged

```python
self.accelerator.gather_for_metrics(chosen_logits.nansum()).nansum().item() / all_num_chosen
```

`.nansum()` reduces over *every* element, so the logged value was the sum of all logits across the whole vocabulary, every sequence position, and the prompt and padding tokens, divided only by the number of sequences. The result scaled with `seq_len × vocab_size` and did not measure a mean logit.

This aligns the computation with `DPOTrainer._compute_loss`: mean over the vocabulary dimension, restricted to completion tokens, token-weighted across ranks. Only `chunk(2, dim=0)` is replaced by `index_select(0, chosen_idx)`, since KTO is unpaired.

## Why not the previous aggregation

The surrounding `nansum() / all_num_*` aggregation is correct for `rewards/*` and `logps/*` and is left untouched: those are per-sequence quantities and KTO ranks hold unequal numbers of chosen/rejected examples, so a plain `gather().mean()` would be biased. `logits/*` is a per-token quantity, so DPO's token-weighted form is both correct and already rank-safe.

---

> [!NOTE]
> **Low Risk**
> Metrics-only change in KTOTrainer._compute_loss; training loss and gradients are unaffected.
> 
> **Overview**
> **KTO** `logits/chosen` and `logits/rejected` were wrong: they summed every element of the full `[batch, seq, vocab]` logits tensor (prompt, padding, and all vocab dims) and divided only by the number of chosen/rejected **sequences**, so logged values scaled with `seq_len × vocab_size` instead of a meaningful mean logit.
> 
> This PR recomputes those metrics like **DPO**: use shifted logits (`shift_logits`), split chosen/rejected with `index_select` on unpaired labels, mask to **completion** tokens only, take **mean over vocab** per token, then **token-weight** sums across ranks via `gather_for_metrics`. `rewards/*` and `logps/*` aggregation is unchanged (still per-sequence `nansum / all_num_*`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2a3188bc71b94b71bf6c9e58d680d35f39bc3cb2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/trainer/kto_trainer.py [hunk 1]
```diff
@@ -1618,6 +1616,25 @@ def _compute_loss(self, model, inputs, return_outputs):
...
+        total_rejected_logits = rejected_logits[rejected_mask.bool()].mean(-1).sum()
+        total_rejected_tokens = rejected_mask.sum()
+        total_chosen_logits = self.accelerator.gather_for_metrics(total_chosen_logits).sum().item()
+        total_chosen_tokens = self.accelerator.gather_for_metrics(total_chosen_tokens).sum().item()
+        total_rejected_logits = self.accelerator.gather_for_metrics(total_rejected_logits).sum().item()
+        total_rejected_tokens = self.accelerator.gather_for_metrics(total_rejected_tokens).sum().item()
+        avg_chosen_logits = total_chosen_logits / total_chosen_tokens if total_chosen_tokens > 0 else 0.0
+        avg_rejected_logits = total_rejected_logits / total_rejected_tokens if total_rejected_tokens > 0 else 0.0
+        self._metrics[mode]["logits/chosen"].append(avg_chosen_logits)
+        self._metrics[mode]["logits/rejected"].append(avg_rejected_logits)
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/trainer/kto_trainer.py]: https://github.com/huggingface/trl/pull/6424/commits/e50ad93485699ab3a2b42049620c19b74c883c5c

[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/trainer/kto_trainer.py]: https://github.com/huggingface/trl/pull/6424/commits/e50ad93485699ab3a2b42049620c19b74c883c5c

**Review by kashif — APPROVED** [review]
